private-templates-service/client/src/components/Common/VersionModal/VersionModal.tsx

```diff
@@ -0,0 +1,160 @@
+  }
+  
+  delete = () => {
+    for (let instance of this.props.template.instances!) {
+      if (!instance.version || !this.state.versionList.includes(instance.version!)) continue;
+      this.props.deleteTemplateVersion(instance.version!);
+      this.props.toggleModal();
+    }
+  }
+
+  publish = () => {
+    for (let instance of this.props.template.instances!){
+      if (!this.state.versionList.includes(instance.version!)) continue;
+      const template = JSON.stringify(instance.json);
+      this.props.updateTemplateState(template, PostedTemplate.StateEnum.Live, instance.version!);
+      this.props.toggleModal();
+    }
+  }
+
+  unpublish = () => {
+    for (let instance of this.props.template.instances!){
+      if (!this.state.versionList.includes(instance.version!)) continue;
+      const template = JSON.stringify(instance.json);
+      this.props.updateTemplateState(template, PostedTemplate.StateEnum.Deprecated, instance.version!);
+      this.props.toggleModal();
+    }
+  }
+
+  updateVersionList = (isChecked?: boolean, version?: string) => {
+    if (!isChecked || !version) return;
+    let versionList = this.state.versionList;
+    if (isChecked === true) {
+      if (versionList.includes(version)) return;
+      versionList.push(version);
+    } else {
+      if (!version.includes(version)) return;
+      let index = versionList.indexOf(version);
+      versionList.splice(index, 1);
+    }
+    this.setState({ versionList: versionList });
+  }
+
+  render() {
+    const { template } = this.props;
+    return (
+      <BackDrop>
+        <Modal>
+          <Header>Versions</Header>
+          <Description>There are {template.instances!.length} versions for <DescriptionAccent>{template.name}</DescriptionAccent></Description>
+          <CenterPanelWrapper>
+            <Card>
+              <CardHeaderRow>
+                <CardHeaderText>Version</CardHeaderText>
+                <CardHeaderText>Published</CardHeaderText>
+                <CardHeaderText>Status</CardHeaderText>
+                <CardHeaderText>{`${this.state.versionList!.length} Selected`}</CardHeaderText>
+              </CardHeaderRow>
+              <CardBody>
+              {this.props.template.instances && this.props.template.instances.map((instance: TemplateInstance) => (
+            <VersionCardRow>
+              <VersionWrapper>
+                {instance.version}
+              </VersionWrapper>      
+              <DateWrapper>{instance.publishedAt? getDateString(instance.publishedAt) : "Not published"}</DateWrapper>
+              <StatusWrapper>
+                <StatusIndicator state={instance.state}/>
+                <Status>{instance.state && instance.state.toString().charAt(0).toUpperCase() + instance.state.toString().slice(1)}</Status>
+              </StatusWrapper>
+              <CheckboxWrapper><Checkbox defaultChecked={false} onChange={(e, checked) => this.updateVersionList(checked, instance.version)}/></CheckboxWrapper>
```

Comment:
I think the "react" way to do this is by making the list of checkboxes all controlled components. That way, we don't need the `updateVersionList` function where we manually keep a secondary list of checked boxes

Reply:
Fixed, thanks!